### PR TITLE
Ns/update serialization doc

### PR DESCRIPTION
**Pull request by nsarlin-zama**
closes: _please link all relevant issues_

### PR content/description
Small reorganization of the serialization doc:
- Give a simple example at the top with `safe_serialize`, so that people can quickly find the recommended way without having to read the full page
- put the serde part at the end, since I consider it more as an advanced mode for users that want complete control

### CODE REVIEW

File: tfhe/docs/fundamentals/serialization.md [hunk 1]
```diff
@@ -108,54 +65,150 @@ use tfhe::{
...
+```
+
+## Serialization/deserialization using serde
+
+**TFHE-rs** uses the [Serde](https://crates.io/crates/serde) framework and implements Serde's `Serialize` and `Deserialize` traits.
+
+This allows you to serialize into any [data format](https://serde.rs/#data-formats) supported by serde. In the following example, we use [bincode](https://crates.io/crates/bincode) for its binary format.
+
+Here is a full example:
+
```
Comment (IceTDrinker) [inline tfhe/docs/fundamentals/serialization.md:154]: maybe explain that TFHE-rs does nothing for the user to guarantee type compatibility here

**Review by IceTDrinker — COMMENTED** [review]
Fine with me, just indicate for the last section that it's a "power user thing" and that TFHE-rs does not provide any safety in that case

**Review by IceTDrinker — COMMENTED** [review]
Good for me, maybe @yuxi-zama will have some comments on that

**Review by IceTDrinker — COMMENTED** [review]
Good for me, maybe @yuxi-zama will have some comments on that

**Review by IceTDrinker — COMMENTED** [review]
Good for me, maybe @yuxi-zama will want to have a look first

**Review by IceTDrinker — COMMENTED** [review]
Good for me, maybe @yuxi-zama will want to have a look first

**Review by IceTDrinker — COMMENTED** [review]
Good for me, maybe @yuxizama will want to have a look first

**Review by yuxizama — APPROVED** [review]
All good for me, thank you!

**Pull request merged**